Tests/SwiftSyntaxTest/MemoryLayoutTest.swift

```diff
@@ -0,0 +1,51 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import XCTest
+@_spi(Testing) import SwiftSyntax
+
+#if DEBUG
+final class MemoryLayoutTest: XCTestCase {
+
+  func testMemoryLayouts() throws {
+    /// This test result is just for tracking the memory footprint of syntax nodes,
```

Comment:
An alternative place we could put this would be the CLI tool, or do you think it's useful to know when they've changed?

Reply:
I think it's useful to know when the size has changed, and what was that for.